drivers/pipes/pipe.c

```diff
@@ -305,9 +303,9 @@ int pipe2(int fd[2], int flags)
 
   /* Clear O_NONBLOCK if it was set previously */
 
-  if ((flags & O_NONBLOCK) == 0)
+  if (!nonblock)
     {
-      ret = fcntl(fd[1], F_SETFL, flags & (~O_NONBLOCK));
+      ret = ioctl(fd[1], FIONBIO, &nonblock);
```

Comment:
Why not reuse `ret` as in `net/local/local_fifo.c`?

Reply:
nonblock has the right type(int) here, but wrong type(bool) in local_tx_open, so the temp assign to ret doesn't need.

Reply:
I mean that in `net/local/local_fifo.c` you have something that can be used here
```
if ((flags & O_NONBLOCK) == 0)
{
  ret = 1;
  ret = ioctl(fd[1], FIONBIO, &ret);
}
```
and get rid of `nonblock`, so my ask was basically why different places implement different approaches?

Reply:
there are many place to check O_NONBLOCK in this function, do you want all place changing to `flags & O_NONBLOCK`?

Reply:
both places use `nonblock`, the only difference is that `nonblock` has different type:

1. nonblock in pipe.c is local variable, so it's fine to change it's type to int to match FIONBIO's expectation
2. nonblock in local_fifo.c is function argument, so it is better to keep bool type

Reply:
yes, but in `file_pipe` and `pipe2` we can get rid of `nonblock` auto variable and just reuse `ret = 1;`.

Reply:
The difference between two is minor, it's fine with either to me:).